semgrep/semgrep/util.py

```diff
@@ -128,3 +130,30 @@ def progress_bar(
         wrapped: Iterable[T] = tqdm(listified, file=file, **kwargs)
         return wrapped
     return listified
+
+
+def sub_run(cmd: List[str], **kwargs: Any) -> Any:
+    """A simple proxy function to minimize and centralize subprocess usage."""
+    # fmt: off
+    result = subprocess.run(cmd, **kwargs)  # nosem: python.lang.security.audit.dangerous-subprocess-use.dangerous-subprocess-use
```

Comment:
It seems like it makes more sense just to remove the rule from the pack, if we effectively don't want to use it on this project at all.

Reply:
Otherwise, it would make sense to me to deliberately disable it on every appearance, rather than hiding it in a wrapper function

Reply:
It's fairly common practice to centralize security-sensitive functionality. This provides a few benefits: 1.) it gives you a central location to harden potentially insecure calls, and 2.) it's easier to audit one location that's been deemed secure than many spread throughout the codebase. This isn't so important here because we have a fairly basic threat model as a CLI app running additional subprocess calls, but it's inline with the experience our users may face. I do think this check is on the fringe of usefulness (a better, less noisy version would be checking for `shell=True`), but again, I think this is a useful exercise in workflows our users will experience.

Additionally, if we deliberately disable on all appearances then we'll also likely have to include a `# fmt: off` and `# fmt: on` because `black` tends to reformat the code such that `semgrep` doesn't recognize the `# nosem`. Adding these three comments around every `subprocess` calls feels like a lot.

Note that, regarding (1) above, `subprocess` calls can take a `str` or a `list`. Using a `list` makes it less likely that string concatenation somewhere leads to arbitrary code execution in a `subprocess` call. So using `mypy` to enforce `cmd` is a `list` is also a benefit here.

Reply:
I'll move forward with this for now and we can always iterate and improve